rmgpy/data/kinetics/kineticsTest.py

```diff
@@ -548,15 +545,30 @@ def testfilterReactions(self):
         
         reactants = np.unique(np.array(reactants)).tolist()
         products = np.unique(np.array(products)).tolist()
+        mreactants = np.array([i.molecule[0] for i in reactants])
+        mproducts = np.array([i.molecule[0] for i in products])
+        lostReactants = mreactants[6:]
+
+        nmreactants = list(set(mreactants)-set(lostReactants))
+        nmproducts = list(set(mproducts)-set(lostReactants))
+
+        out = filterReactions(nmreactants,nmproducts,reactions)
+            
+        rset = list(set(reactions) - set(out))
+
+        msets = [set(i.reactants+i.products) for i in rset]
+        lrset = set(reactants[6:])
         
-        out = []
-        for i in reactants:
-            out += filterReactions([i],[],reactions)
-        for j in products:
-            out += filterReactions([],[i],reactions)
+        for i in range(len(msets)): #test that all the reactions we removed are missing a reactant or a product
+            iset = msets[i]
+            self.assertTrue(iset & lrset != set(),msg='reaction {0} removed improperly'.format(rset[i]))
+        
+        outsets = [set(i.reactants+i.products) for i in out]
             
-        for i in out:
-            self.assertIn(i,reactions, 'filter reactions is removing some reaction/s regardless of reactant or product input')
+        for i in range(len(out)): #test that all the reactions left in aren't missing any reactants or products
+            iset = outsets[i]
+            self.assertFalse(iset & lrset != set(),msg='reaction {0} left in improperly, should have removed in based on presence of {1}'.format(out[i],iset & lrset))
```

Comment:
Perhaps `assertTrue` that `iset & lrset == set()` instead? It feels a bit more intuitive to avoid a double negative.